hanson_at_fb.py

```diff
@@ -0,0 +1,33 @@
+# given a list of strings, determine which are anagrams and group them together.
+# return a list of lists of anagrams
+
+# input ['cat', 'act', 'dog', 'god', 'def']
+# output [['cat', 'act'], ['dog', 'god'], ['def']]
+
+# lists are assigned by reference, so list1 = list2, list 1 points to list 2
+# to clone a list, you want to do list1 = list(list2)
+
+# sorted(string) = list of chars of string
+# sort(list) = modifies list in-place
+# ''.join(list), joins each element of the list together with '' (no space) in between
+
+
+def find_anagrams (some_list):
+	grouped_words = {}
+	for word in some_list:
+		sorted_word = ''.join(sorted(word))
+		print sorted_word
+		#if grouped_words[sorted_word]:
+		if sorted_word in grouped_words:
```

Comment:
so to summarize about the defaultdict datastructure:
basically, whenever a defaultdict encounters a unique key, it will call whatever function you passed into the constructor. so what do i want it to do every time it counters a new key? in this case, i want it to instantiate an empty list object, thus I will pass in 'list', which stands for list() with no arguments passed to it, which is the constructor for an empty list object

for example if i want a dictionary of lists, where the value of each dictionary key is a list, i would do
x = defaultdict(list)
so now every time the defaultdict named 'x' encounters a unique key, it will call list(), which is the constructor for an empty list.

but I could also do a dictionary of sets:
x = defaultdict(set)

or a dictionary of ints:
x = defaultdict(int)

this eliminates the need to check for the existence of a key with if statements before operating on it.